src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface_kernels.jl

```diff
@@ -49,10 +49,10 @@ const μ = 1.0 - δ - γ - ϵ
 
 # Enforce NoFlux conditions for `η★`
 
-@inline δxᶠᵃᵃ_η(i, j, k, grid, ::Type{Bounded},        η★::Function, args...) = ifelse(i == 1, 0.0, δxᶠᵃᵃ(i, j, k, grid, η★, args...))
-@inline δyᵃᶠᵃ_η(i, j, k, grid, ::Type{Bounded},        η★::Function, args...) = ifelse(j == 1, 0.0, δyᵃᶠᵃ(i, j, k, grid, η★, args...))
-@inline δxᶠᵃᵃ_η(i, j, k, grid, ::Type{RightConnected}, η★::Function, args...) = ifelse(i == 1, 0.0, δxᶠᵃᵃ(i, j, k, grid, η★, args...))
-@inline δyᵃᶠᵃ_η(i, j, k, grid, ::Type{RightConnected}, η★::Function, args...) = ifelse(j == 1, 0.0, δyᵃᶠᵃ(i, j, k, grid, η★, args...))
+@inline δxᶠᵃᵃ_η(i, j, k, grid, ::Type{Bounded},        η★::Function, args...) = ifelse(i == 1, zero(grid), δxᶠᵃᵃ(i, j, k, grid, η★, args...))
+@inline δyᵃᶠᵃ_η(i, j, k, grid, ::Type{Bounded},        η★::Function, args...) = ifelse(j == 1, zero(grid), δyᵃᶠᵃ(i, j, k, grid, η★, args...))
+@inline δxᶠᵃᵃ_η(i, j, k, grid, ::Type{RightConnected}, η★::Function, args...) = ifelse(i == 1, zero(grid), δxᶠᵃᵃ(i, j, k, grid, η★, args...))
+@inline δyᵃᶠᵃ_η(i, j, k, grid, ::Type{RightConnected}, η★::Function, args...) = ifelse(j == 1, zero(grid), δyᵃᶠᵃ(i, j, k, grid, η★, args...))
```

Comment:
Have you defined `zero(grid)` as the zero element of the eltype on the grid? Because I assume this to return a whole grid with zeros in it, similar as to how
```julia
julia> zero(rand(2,2))
2×2 Matrix{Float64}:
 0.0  0.0
 0.0  0.0
```
As far as I understand these functions (because of `@inline` and provided indices they are applied to all elements in an array), it's probably wise to directly pass on the zero in the correct type to avoid many `convert(T,0)` though!

Reply:
`zero(grid)` returns `zero(eltype(grid))`. But maybe this is poor design. We use the pattern a lot so its nice to make it a little more readable...

I don't think there's a purpose to a grid full of zeros, or at least it's unclear exactly what that means, since we can't use a grid in arithmetic

Reply:
There's also the `zeros` function which could be used  as `zeros(::Type{<:AbstractGrid},n,m,...)` in order to create a grid full of zeros. I think the `zero` function is technically meant to give for any type the zero element so that `A + zero(A) == A` and `A * zero(A) == zero(A)`

Reply:
hm... seems you are right

```Julia
help?> zero
search: zero zeros iszero set_zero_subnormals get_zero_subnormals count_zeros RoundToZero leading_zeros trailing_zeros RoundFromZero finalizer UndefInitializer

  zero(x)
  zero(::Type)


  Get the additive identity element for the type of x (x can also specify the type itself).

  See also iszero, one, oneunit, oftype.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> zero(1)
  0

  julia> zero(big"2.0")
  0.0

  julia> zero(rand(2,2))
  2×2 Matrix{Float64}:
   0.0  0.0
   0.0  0.0
```

so perhaps we need to re-think about `zero(grid)` behavior (in a different PR tho). @glwagner what do you reckon?